test/integration/analyze_test.dart

```diff
@@ -130,6 +133,21 @@ void main() {
       expect(analysis.exitCode, equals(0));
     });
 
+    test('should report lints as fatal if configured to do so', () async {
+      Analysis analysis =
+          await analyzeProject(projectWithLints, fatalLints: true);
+      expect(analysis.numErrors, equals(1));
+      expect(analysis.numHints, equals(0));
+      expect(analysis.exitCode, greaterThan(0));
+    });
+
+    test('should not report hints as fatal if none existed', () async {
```

Comment:
should this be `should not report lints as fatal if none existed`

as well should you check to make sure that the numErrors are 0